tools/oeapkman/oeapkman

```diff
@@ -0,0 +1,396 @@
+        if ! command -v curl > /dev/null; then
+            if ! command -v wget > /dev/null; then
+                echo "Install curl or wget to proceed." && exit 1
+            fi
+        fi
+
+        if command -v curl > /dev/null; then
+            curl -L -o alpine-minirootfs.tar.gz "$ALPINE_ROOTFS_URL"
+        elif command -v wget > /dev/null; then
+            wget -O alpine-minirootfs.tar.gz  "$ALPINE_ROOTFS_URL"
+        else
+            echo "Install curl or wget to proceed." && exit 1
+        fi
+
+        sh -c "mkdir -p tmp && cd tmp && tar xf ../alpine-minirootfs.tar.gz"
+        cp /etc/resolv.conf tmp/etc/resolv.conf
+        echo "https://dl-cdn.alpinelinux.org/alpine/edge/testing" >> tmp/etc/apk/repositories
+        mkdir tmp/apkbuild
+        mv tmp "${ALPINE_FS}"
+
+        echo "Checking if unshare works..."
+        if ! alpine_exec echo "unshare works" 2>/dev/null; then
+            # Download and use proot
+            echo "unshare doesn't work"
+            if command -v curl > /dev/null; then
+                curl -L -o proot.zip "$PROOT_URL"
+            else
+                wget -O proot.zip "$PROOT_URL"
+            fi
+            "$ALPINE_FS/lib/ld-musl-x86_64.so.1" "$ALPINE_FS/bin/busybox" unzip proot.zip
+            MODE="proot"
+        else
+            MODE="unshare"
+        fi
+
+        # Initialize apk
+        alpine_exec sh -c "apk update && apk add bash"
+        echo "oeapkman configured to use \`$MODE\`"
+    fi
+}
+
+alpine_exec()
+{
+    export SHELL=/bin/sh
+    export TERM=xterm-256color
+    export PS1="oeapkman \w$ "
+    export PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin
+
+    if test -f dist/proot; then
+        # Root home folder is /root. Therefore bind both $HOME and /home
+        dist/proot -b /dev -b /dev/pts -b /proc -b /sys -b /dev/shm \
+                   -b "$HOME" \
+                   -b /home \
+                   -b /opt \
+                   -r alpine-fs \
+                   -0 \
+                   -w "$CURRENT_DIR" \
+                    "$@"
+
+    else
+        ALPINE_EXEC_CMD="
+            mount --rbind /home \"$ALPINE_FS/home\"
+            mount --rbind /opt \"$ALPINE_FS/opt\"
+            mount -t proc /proc \"$ALPINE_FS/proc\"
+            mount --rbind /dev \"$ALPINE_FS/dev\"
+            mount --rbind  /sys \"$ALPINE_FS/sys\"
+            chroot \"$ALPINE_FS\" sh -c 'cd \"$CURRENT_DIR\" && \"\$@\"' oeapkman \"\$@\"
+        "
+        unshare --map-root-user --pid --fork --mount --propagation unchanged \
+                sh -c "$ALPINE_EXEC_CMD" oeapkman "$@"
+    fi
+}
+
+
+setup_alpine_fs
+
+wrap_ld()
+{
+    if test -f "$ALPINE_FS/apkbuild/ld.wrap.done"; then
+        return
+    fi
+    echo "Wrapping ld"
+    LD_DIR="${ALPINE_FS}/usr/x86_64-alpine-linux-musl/bin"
+    if ! test -f "$LD_DIR/ld.orig"; then
+        mv "$LD_DIR/ld" "$LD_DIR/ld.orig"
+    fi
+    # Remove ld found in usr/bin folder so that everything get routed to
```

Comment:
```suggestion
    # Remove ld found in the usr/bin folder so that everything gets routed to
```
grammar

Reply:
Removed advanced functionality. This section is no longer there.